### PR TITLE
Support for the wt150 printer (mightyboard reve) with RepRapDiscount Full Graphic Smart Controller

### PR DESCRIPTION
**Pull request by stawel**
The Weistek wt150 printer is a cheap Chinese printer based on mightyboard reve board
with a DMT32240T035_02WT touch display.
The original display is not supported by this pull request,
however it is possible to connect a RepRapDiscount Full Graphic Smart Controller to it.

This pull request is only a proposition I would like to know what you think,
especially the last commit https://github.com/stawel/Marlin/commit/b7cc5a036ed9aae2af2abe2dfa3e37a7a18905e1
is just a hotfix for some issues I've encountered:
1. error when TEMP_0_PIN is not defined (mightyboard uses a MAX6675 SPI based converter)
2. "//lcd_wait_for_homing();" - in "prepare"->"Level bed"->"Level bed" printer stops after Z homing, display shows "Homing XYZ" indefinitely

### CODE REVIEW

File: Marlin/ultralcd.cpp [hunk 1]
```diff
@@ -1298,7 +1298,7 @@ void kill_screen(const char* lcd_msg) {
         #endif
       ;
       if (no_reentrance) return;
-      lcd_wait_for_homing();
+      //lcd_wait_for_homing();
```
Comment (thinkyhead) [inline Marlin/ultralcd.cpp:1301]: This workaround isn't needed anymore. I patched the issue earlier.

**Comment by thinkyhead** [comment]
This is a really great contribution. You know, up to now we haven't even had the Mightyboard working at all, so this is a huge step.

Do you know if this means Marlin can work on old MB Replicators too? When I studied the Rev.E schematics it seemed apparent that we would need to be able to have multiple i2c buses, so I'm guessing we are still far from that goal.

**Comment by stawel** [comment]
> Do you know if this means Marlin can work on old MB Replicators too?

As far as I'm aware of it should work, I'm using the Mightyboard schematics and I didn't noticed any significant differences:
1. wt150: https://github.com/stawel/stuff/blob/master/ideawerk_3d_printer_wt150/pics/IMG_20150815_114118.jpg
2. Mightyboard: http://www.thingiverse.com/thing:16058 

the only not implemented feature is currently a second max6675 for dual extrusion.

>When I studied the Rev.E schematics it seemed apparent that we would need to be able to have multiple i2c buses

Yes, I used an external lib for that (I'm not sure if this is ok): https://github.com/felias-fogg/SlowSoftI2CMaster
https://github.com/MarlinFirmware/Marlin/pull/5670/commits/30cd4254f5f399481f3fb5f377f68593ea11738f#diff-15609c3ccbe545a6a2e140622c59b2aaR59

Also controversial ;), there was a need for a software spi for max6675 communication, so I've added: https://github.com/niteris/ArduinoSoftSpi to the project: 
https://github.com/MarlinFirmware/Marlin/pull/5670/commits/8d3e5caf28a59a4b0548ce449ed661d821b3f1a1#diff-634447c3c51721b45c1da702b1805034R1
and a generic Marlin/spi.h template. 

And finally, I do some ugly pin count checks:
https://github.com/MarlinFirmware/Marlin/pull/5670/commits/9b3019ebc0b8d1ba9cf08c2621f23d06465bb38a#diff-aa37f1d089c5719aac8c0a936e2ea9f7R258
I couldn't come up with anything better.

> so I'm guessing we are still far from that goal.

not really,

**Comment by stawel** [comment]
I just rebased my fork to the newest RCBugFix commit (pushed the rebase with -f to github)
but I still need this hotfix:
"//lcd_wait_for_homing();" - in "prepare"->"Level bed"->"Level bed" printer stops after Z homing, display shows "Homing XYZ" indefinitely

**Comment by thinkyhead** [comment]
@stawel I've rebased this and moved it to a new PR at #6284. Let's continue there.

I made one tweak, which was to change the sanity check to what we really want: allow a board with no thermistor if it defines a thermocouple.

**Pull request closed without merging**